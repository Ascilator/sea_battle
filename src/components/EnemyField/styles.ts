import styled from 'styled-components';

export const StyledFieldCont = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
`;

export const StyledField = styled.div`
  width: 40vw;
  height: 40vw;
  user-select: none;
`;

export const StyledCoords = styled.div`
  display: flex;
  align-items: stretch;
  width: 100%;
  height: calc(100% / 12);
`;

export const StyledCell = styled.div`
  flex: 0 1 calc(100% / 12);
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
`;
