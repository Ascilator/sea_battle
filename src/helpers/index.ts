export * from './socket';
export * from './components';
export * from './testHelper';
