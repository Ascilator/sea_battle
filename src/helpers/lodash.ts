export const deepClone = (array: Array<any>): Array<any> => {
  const result: Array<any> = [];

  array.forEach(el => {
    if (!Array.isArray(el)) {
      result.push(el);
    } else {
      result.push(deepClone(el));
    }
  });

  return result;
};
