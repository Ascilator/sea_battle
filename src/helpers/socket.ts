import io from "socket.io-client";

import { BASE_URL_FOR_SOCKETS } from "@/constants";

const RECONNECT_ATTEMPTS: number = 2;

const socket = io(BASE_URL_FOR_SOCKETS, {
  path: `${""}/socket.io/`,
  transports: ["websocket"],
  reconnection: true,
  reconnectionAttempts: RECONNECT_ATTEMPTS,
  reconnectionDelay: 1000,
  reconnectionDelayMax: 5000,
  autoConnect: true,
});

export { socket };
