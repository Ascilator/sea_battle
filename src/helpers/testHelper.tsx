import { render as rtlRender } from '@testing-library/react';
import { FC, ReactElement } from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';

export const renderComponentForTest = (
  component: ReactElement,
  {
    store,
    ...renderOptions
  }: {
    store: Store;
  }
) => {
  //  <Provider store={store}>

  const Wrapper: FC = ({ children }) => <Provider store={store}>{children}</Provider>;

  return rtlRender(component, { wrapper: Wrapper, ...renderOptions });
};

export * from '@testing-library/react';
