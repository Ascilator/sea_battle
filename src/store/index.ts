import { configureStore } from '@reduxjs/toolkit';

import canClickReducer from './canClick';

export const store = configureStore({
  reducer: {
    canClick: canClickReducer
  }
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
