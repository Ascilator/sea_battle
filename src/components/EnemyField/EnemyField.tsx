import { FC, useState } from 'react';
import { alphabet, HIT, KILL_THE_SHIP, matrix, MISS } from '@/constants';
import { FieldState } from '@/types';
import { FieldRow } from '@/components/FieldRow';
import { useAppDispatch, useSocketListeners } from '@/hooks';
import { changeMatrix, killShip } from '@/helpers';
import { changeTurnByData } from '@/store/turn';
import { deepClone } from '@/helpers/lodash';

import { StyledCoords, StyledCell, StyledFieldCont, StyledField } from './styles';
import { DefShotData } from '../FieldRow/types';

const EnemyField: FC = () => {
  const [gameState, setGameState] = useState<FieldState>(matrix);
  const dispatch = useAppDispatch();

  useSocketListeners([
    {
      eventName: MISS,
      callback: ({ x, y }: { x: number; y: number }) => {
        setGameState(prevState => changeMatrix(x, y - 1, prevState, 7));
      }
    },
    {
      eventName: HIT,
      callback: ({ x, y }: { x: number; y: number }) => {
        setGameState(prevState => changeMatrix(x, y - 1, prevState, 6));
        dispatch(changeTurnByData(true));
      }
    },
    {
      eventName: KILL_THE_SHIP,
      callback: ({ x, y }: DefShotData) => {
        setGameState(prevState => {
          killShip(prevState, x, y);
          return deepClone(prevState);
        });
      }
    }
  ]);

  const renderRows = () =>
    gameState.map((row, index) => <FieldRow key={index} value={index + 1} rowData={row} enemy />);

  const renderCoords = () => (
    <StyledCoords>
      <StyledCell />
      {alphabet.map(el => (
        <StyledCell key={el}>{el}</StyledCell>
      ))}
      <StyledCell />
    </StyledCoords>
  );

  return (
    <StyledFieldCont>
      <StyledField>
        {renderCoords()}
        {renderRows()}
        {renderCoords()}
      </StyledField>
    </StyledFieldCont>
  );
};

export { EnemyField };
