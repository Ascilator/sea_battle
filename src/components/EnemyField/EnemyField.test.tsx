import { renderComponentForTest } from '@/helpers';
import { store } from '@/store';

import { EnemyField } from './EnemyField';

test('EnemyField component', () => {
  const enemyFiled = renderComponentForTest(<EnemyField />, {
    store
  });

  expect(enemyFiled).toMatchSnapshot();
});
