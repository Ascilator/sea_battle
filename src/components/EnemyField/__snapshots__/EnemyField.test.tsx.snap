// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnemyField component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKwDye eubgr"
      >
        <div
          class="sc-eCImPb fEIBIZ"
        >
          <div
            class="sc-jRQBWg VAgvn"
          >
            <div
              class="sc-gKclnd HCazm"
            />
            <div
              class="sc-gKclnd HCazm"
            >
              a
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              b
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              c
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              d
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              f
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              g
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              h
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              i
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              j
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              k
            </div>
            <div
              class="sc-gKclnd HCazm"
            />
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              1
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              1
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              2
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              2
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              3
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              3
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              4
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              4
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              5
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              5
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              6
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              6
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              7
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              7
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              8
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              8
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              9
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              9
            </div>
          </div>
          <div
            class="sc-gsDKAQ bcactL"
          >
            <div
              class="sc-dkPtRN ePBYqE"
            >
              10
            </div>
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-bdvvtL kKXOVj"
              color="#122FAA"
            />
            <div
              class="sc-dkPtRN ePBYqE"
            >
              10
            </div>
          </div>
          <div
            class="sc-jRQBWg VAgvn"
          >
            <div
              class="sc-gKclnd HCazm"
            />
            <div
              class="sc-gKclnd HCazm"
            >
              a
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              b
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              c
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              d
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              f
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              g
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              h
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              i
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              j
            </div>
            <div
              class="sc-gKclnd HCazm"
            >
              k
            </div>
            <div
              class="sc-gKclnd HCazm"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKwDye eubgr"
    >
      <div
        class="sc-eCImPb fEIBIZ"
      >
        <div
          class="sc-jRQBWg VAgvn"
        >
          <div
            class="sc-gKclnd HCazm"
          />
          <div
            class="sc-gKclnd HCazm"
          >
            a
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            b
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            c
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            d
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            f
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            g
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            h
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            i
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            j
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            k
          </div>
          <div
            class="sc-gKclnd HCazm"
          />
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            1
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            1
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            2
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            2
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            3
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            3
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            4
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            4
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            5
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            5
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            6
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            6
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            7
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            7
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            8
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            8
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            9
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            9
          </div>
        </div>
        <div
          class="sc-gsDKAQ bcactL"
        >
          <div
            class="sc-dkPtRN ePBYqE"
          >
            10
          </div>
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-bdvvtL kKXOVj"
            color="#122FAA"
          />
          <div
            class="sc-dkPtRN ePBYqE"
          >
            10
          </div>
        </div>
        <div
          class="sc-jRQBWg VAgvn"
        >
          <div
            class="sc-gKclnd HCazm"
          />
          <div
            class="sc-gKclnd HCazm"
          >
            a
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            b
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            c
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            d
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            f
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            g
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            h
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            i
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            j
          </div>
          <div
            class="sc-gKclnd HCazm"
          >
            k
          </div>
          <div
            class="sc-gKclnd HCazm"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
